functions/datastore/test/index.test.js

```diff
@@ -1,310 +1,266 @@
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//    http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+/**
+ * Copyright 2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
 'use strict';
 
-var proxyquire = require('proxyquire').noCallThru();
+const proxyquire = require(`proxyquire`).noCallThru();
 
-var KEY = 'key';
-var KIND = 'user';
+const NAME = `sampletask1`;
+const KIND = `Task`;
+const VALUE = {
+  description: `Buy milk`
+};
 
 function getSample () {
-  var datastore = {
-    delete: sinon.stub().callsArg(1),
-    get: sinon.stub().callsArg(1),
-    key: sinon.stub().returns({
-      kind: KIND,
-      path: KEY
-    }),
-    save: sinon.stub().callsArg(1)
+  const key = {
+    kind: KIND,
+    name: NAME,
+    path: [KIND, NAME]
+  };
+  const entity = {
+    key: key,
+    data: VALUE
+  };
+  const datastore = {
+    delete: sinon.stub().returns(Promise.resolve()),
+    get: sinon.stub().returns(Promise.resolve([entity])),
+    key: sinon.stub().returns(key),
+    save: sinon.stub().returns(Promise.resolve())
   };
-  var DatastoreMock = sinon.stub().returns(datastore);
+  const DatastoreMock = sinon.stub().returns(datastore);
+
   return {
-    sample: proxyquire('../', {
+    program: proxyquire(`../`, {
       '@google-cloud/datastore': DatastoreMock
     }),
     mocks: {
       Datastore: DatastoreMock,
-      datastore: datastore
+      datastore: datastore,
+      key: key,
+      entity: entity,
+      req: {
+        body: {
+          kind: KIND,
+          key: NAME,
+          value: VALUE
+        }
+      },
+      res: {
+        status: sinon.stub().returnsThis(),
+        send: sinon.stub().returnsThis()
+      }
     }
   };
 }
 
-function getMockContext () {
-  return {
-    success: sinon.stub(),
-    failure: sinon.stub()
-  };
-}
+describe(`functions:datastore`, () => {
+  it(`set: Set fails without a value`, () => {
+    const expectedMsg = `Value not provided. Make sure you have a "value" property in your request`;
```

Comment:
I feel like the general pattern I follow is to pull the expected value into a variable when the line is getting long.